pghoard/transfer.py

```diff
@@ -309,25 +310,27 @@ def handle_download(self, site, key, file_to_transfer):
             self.metrics.unexpected_exception(ex, where="handle_download")
             return CallbackEvent(success=False, exception=ex, opaque=file_to_transfer.opaque)
 
-    def handle_upload(self, site, key, file_to_transfer):
+    def handle_upload(self, site, key, file_to_transfer: UploadEvent):
         payload = {"file_size": file_to_transfer.file_size}
         try:
             storage = self.get_object_storage(site)
             unlink_local = file_to_transfer.remove_after_upload
             self.log.info("Uploading file to object store: src=%r dst=%r", file_to_transfer.source_data, key)
             if not isinstance(file_to_transfer.source_data, BytesIO):
-                f = open(file_to_transfer.source_data, "rb")
+                f = open(file_to_transfer.source_data, "rb")  # type: ignore
             else:
                 f = file_to_transfer.source_data
             with f:
                 metadata = file_to_transfer.metadata.copy()
                 if file_to_transfer.file_size:
-                    metadata["Content-Length"] = file_to_transfer.file_size
-                storage.store_file_object(key, f, metadata=metadata)
+                    metadata["Content-Length"] = file_to_transfer.file_size  # type: ignore
```

Comment:
Here too, `if file_to_transfer.file_size is not None` should remove the need to disable type checking.